Adding E paper display capability

JodliDev opened the pull request:
I added the GXEPD2 library and incorporated it.
For now there is only the option for the TTGOv5 2.3 (because that is the board we have).
But by changing **src/hal/ttgov5_v2_3.h** a lot of other boards can be added (see https://github.com/ZinggJM/GxEPD2 and https://github.com/ZinggJM/GxEPD2/blob/master/examples/GxEPD2_Example/GxEPD2_display_selection_new_style.h).

Also the screen does not have many options at the moment. Adapt **src/e_paper_display.cpp** to add capabilites.

The screen behavior is slightly different than with OLED screens, because e paper displays do not need constant updates and it takes a few seconds to update (and I didnt see any point in updating it regularly since that just wastes battery).
So the screen only updates when: The IO button is pressed or the board wakes up from deep sleep.

cyberman54 commented on platformio_orig.ini:
```diff
@@ -65,6 +66,7 @@ display_library = ; set by build.py and taken from hal file
 lib_deps_lora =
     mcci-catena/MCCI LoRaWAN LMIC library @ ^3.3.0
 lib_deps_display =
+    https://github.com/lewisxhe/GxEPD.git
```
Shouldn't we better use [GxEPD2](https://github.com/ZinggJM/GxEPD2) here?

JodliDev replied:
Ah. I am sorry. That is a remanence from my testing with another library that is specifically targeted for ttgo5. I changed it in my plattformio.ini but apparantly forgot to change it in platformio_orig.ini

cyberman54 commented on src/irqhandler.cpp:
```diff
@@ -52,6 +52,14 @@ void irqHandler(void *pvParameters) {
 
+// E-Ink display needs refresh?
+#ifdef HAS_E_PAPER_DISPLAY
+if (InterruptStatus & E_PAPER_DISPLAY_IRQ) {
```
Where is E_PAPER_DISPLAY_IRQ set? Can't find a trigger. It looks like display is only refreshed when the button is pressed, but not by a cyclic IRQ. Thus, we don't need this?

JodliDev replied:
Yes you are right. I did not completely understand what this was for and thought it might be important for other things and also thought you might want it at a later time for automatic screen updates (which we dont need in our project)

cyberman54 reviewed (changes requested):
See comment in irqhandler.h, the irq source for refreshing the e-paper-display is missing?

JodliDev commented:
I changed the source (sorry for that) and removed the irq handler for now.
I can add automatic updates if you prefer, but I need to wait for a couple of days until I can test it (right now I am doing a battery test with our board).

cyberman54 commented on include/irqhandler.h:
```diff
@@ -34,6 +34,10 @@ void IRAM_ATTR DisplayIRQ();
 void IRAM_ATTR MatrixDisplayIRQ();
 #endif
 
+#ifdef HAS_E_PAPER_DISPLAY
```
we don't need this here.

cyberman54 reviewed (changes requested):
Please see comment in code.

JodliDev commented:
Sorry for my late response. Yes, irqhandler.h does not need an include anymore. But button.h does. So I moved the include
